sdk/src/main/scala/com/horizen/block/SidechainBlockBase.scala

```diff
@@ -146,17 +149,25 @@ abstract class SidechainBlockBase[TX <: Transaction, H <: SidechainBlockHeaderBa
         throw new InvalidSidechainBlockDataException(s"${getClass.getSimpleName} $id MainchainHeader ${mainchainHeaders(i).hashHex} is not a parent of MainchainHeader ${mainchainHeaders(i+1)}.")
     }
 
+    var totalTxSize = 0L
     // Check that SidechainTransactions are valid.
     for(tx <- sidechainTransactions) {
       Try {
         tx.semanticValidity()
+        totalTxSize = totalTxSize + tx.size()
       } match {
         case Success(_) =>
         case Failure(e) => throw new InvalidSidechainBlockDataException(
           s"${getClass.getSimpleName} $id Transaction ${tx.id} is semantically invalid: ${e.getMessage}.")
       }
     }
 
+    // Check we do not exceed the block overhead size
+    val blockOverheadSize = blockSize-totalTxSize
```

Comment:
here we consider totalTxSize as a sum of tx bytes, but we do not consider `TxCompanion` and `List`  overhead for the Txs.
In UTXO forger we have rough estimation:
`val txSize = tx.bytes.length + 4 // placeholder for Tx length`
But here we need to be more precise (so may get lower value):
let's introduce another method like: `protected blockTxsSize(): Long`
that on Account and UTXO block will use `companion` to serialize the list of sidechain transactions.

Reply:
Ok, done